pennylane/qnn/torch.py

```diff
@@ -207,7 +210,35 @@ def __init__(self, qnode, weight_shapes: dict, init_method: Optional[Callable] =
             )
         super().__init__()
 
-        self.sig = qnode.func.sig
+        weight_shapes = {
+            weight: (tuple(size) if isinstance(size, Iterable) else (size,) if size > 1 else ())
+            for weight, size in weight_shapes.items()
+        }
+
+        # validate the QNode signature, and convert to a Torch QNode.
+        if qml.tape_mode_active():
+            self._signature_validation_tape_mode(qnode, weight_shapes)
+            self.qnode = qnode
+            self.qnode.to_torch()
+        else:
+            self._signature_validation(qnode, weight_shapes)
+            self.qnode = to_torch(qnode)
```

Comment:
In tape mode, the signature validation and method of applying an interface differs, so we must branch here.